runtime/src/gpu/amd/gpu-amd.c

```diff
@@ -512,5 +512,60 @@ void chpl_gpu_impl_host_unregister(void* var) {
   ROCM_CALL(hipHostUnregister(var));
 }
 
+void chpl_gpu_impl_name(int dev, char **result) {
+  char* name = (char *)chpl_mem_alloc(0xFF, CHPL_RT_MD_IO_BUFFER, __LINE__, 0);
```

Comment:
Why `0xFF` for size?

Reply:
I have a preference towards doing this allocation in `chpl-gpu.c` as I don't see this as a vendor-specific operation.

Reply:
> Why 0xFF for size?

I need to have some cap on the size and that seems like a nice, round, number that won't waste too much memory but will be large enough to store any reasonable device name.

I've factored out the (non GPU-vendor specific) allocation into `chpl-gpu.c`.